editor/editor_inspector.cpp

```diff
@@ -1595,7 +1595,7 @@ void EditorInspector::update_tree() {
 			if (capitalize_paths)
 				cat = cat.capitalize();
 
-			if (!filter.is_subsequence_ofi(cat) && !filter.is_subsequence_ofi(name))
+			if (!filter.is_subsequence_ofi(cat) && !filter.is_subsequence_ofi(name) && property_prefix.to_lower().find(filter.to_lower()) == -1)
```

Comment:
I wonder why this is needed? The previous code was already checking if the `filter` is either in the property name or the rest of its path (`cat`), so here it should have been working fine?

Reply:
Judging by the code, it's not meant to be empty, otherwise there wouldn't be a point in testing it. So yeah, would be worth investigating deeper.

Reply:
Seems like when the `EditorInspector` is used in the engine Inspector `cat` is correctly set and `property_prefix` is empty, viceversa when it's used either in the project or editor settings.

Those variable are deduced from the `EditorInspector`'s property list on [line 1463](https://github.com/godotengine/godot/pull/33047/files/b03f5bb99d409cb5f369d04e08ba0541e6538cba#diff-554c9c97362d5a6f63b7244253542ffeL1463) and it has been this way since [`005b69cf6e`](https://github.com/godotengine/godot/commit/005b69cf6e276209464cc8c36ebc7376679925b6) when `editor_inspector.h/cpp` have been created, so my guess is that it's expected to be like so. Don't know if I'm missing something.